### ISSUE TITLE
Pollution and server population

### ISSUE THREAD

**Issue by Thyriel** [StrangeLoopGames/EcoIssues#5179]
Not sure how good pollution is already implemented but as it is now it definitly needs some more tweaks:

We play with ~4-5 active players on a 2km² map but so far even in endgame we did not manage yet to create any noticeable pollution. The pollution overlay only shows it directly above smelteries (and some other buildings that create them) but nowhere else. Even my wheat fields directly (maybe 10 blocks) beneath of our main smeltery with 4 blast furnaces running almost all the time still show 100% pollution conditions when inspecting with soil sampler. So currently it seems impossible to make any noticeable pollution effect, not to speak of the promised ecological disaster or even noticeable CO2 increase (over a week of playing in mid/endgame)

But thinking a bit ahead, there will be people playing the game solo, other playing with a few active players and even others maybe somewhen running servers with 50-100 people and all of them should have around the same pollution problems regardless if they play with only a few players or with a lot.

So there's two things that should be done in my opinion:
- further tweak pollution effects so they are generating noticeable impacts on the ecosystem (doesn't seem to work currently)
- add a config file for pollution effects that allows to balance player amount vs. pollution effects for server admins (maybe integrate in difficulty.eco ?)

**Comment by UnknownViper** [comment]
Considering the size of your world and amount of players in it, it makes sense. Each and every plant and tree provides an air pollution clearing value, which is naturally amplified by the size of your world.
Then add to it that 4-5 people don't pollute all that much, which means you're never really going to see much of any air pollution.

If you check the pipes tab on your blast furnace and hover over the right-most icon, you will get information about the current pollution, as well as how much is cleared per day (as long as the blast furnace is currently producing something).
It should also be noted that air pollution appears to be global, so it doesn't matter if for instance you have a lot of trees close by. As long as they're in the world, they'll help with clearing air pollution.

Long story short: Your problem is the size of the world you selected for the amount of players you are.
If 50-100 players were in the same world, the situation would be different, as they're be a higher need for items produced in air polluting objects, as well as more people claiming land and damaging the environment, meaning less plants/trees to clear air pollution.

You can already change the pollution values. For instance in EcoSim.eco you can control how much pollution each type of plant or tree clears, as well as what some other things produce (like animal carcasses).
For objects, like the blast furnace, it appears that it's done for each object through the \Mods\AutoGen\WorldObject\ .eco files. This is not something I've tried doing myself yet, so I can't say for sure how to do it.

**Comment by Thyriel** [comment]
> Long story short: Your problem is the size of the world you selected for the amount of players you are.

That's why the game needs easy to configure global variables like multipliers for pollution / air clearing from plants. There will always be people that want bigger worlds with less people. Or you start with 20 active players and end up with 5 after a few days and so on.
Configuring each plant one by one isn't a good solution here

**Comment by UnknownViper** [comment]
That is true. I'm actually working on a tool to help configure some of this stuff a bit easier, but it could certainly be ideal if there was a sort of global "pollution factor" that would scale all things accordingly.

**Comment by BlazeGuy2** [comment]
This issue was moved to StrangeLoopGames/EcoSuggestions#228
